packages/playwright-core/src/server/chromium/videoRecorder.ts

```diff
@@ -160,9 +176,32 @@ export class VideoRecorder {
   async stop() {
     if (this._isStopped)
       return;
-    this.writeFrame(Buffer.from([]), this._lastFrameTimestamp + (monotonicTime() - this._lastWriteTimestamp) / 1000);
+
     this._isStopped = true;
+
+    if (this._firstFrameTimestamp === null || !this._lastFrameBuffer) {
+      if (this._gracefullyClose)
```

Comment:
The callback should never be null at this point. We didn't have this check before, why do you need it now?